### ISSUE TITLE
an error

### ISSUE THREAD

**Issue by sun-fox-cj** [rs/SDWebImage#1440]
there are some error message in console when I call "- (void)sd_setImageWithURL:(NSURL *)url placeholderImage:(UIImage *)placeholder completed:(SDWebImageCompletionBlock)completedBlock;" method, I think you should check nil in file "SDWebImageDecoder.m"
![](http://7xpsbt.com1.z0.glb.clouddn.com/isNil.png)
![](http://7xpsbt.com1.z0.glb.clouddn.com/console.png)
```
+ (UIImage *)decodedImageWithImage:(UIImage *)image {
    // while downloading huge amount of images
    // autorelease the bitmap context
    // and all vars to help system to free memory
    // when there are memory warning.
    // on iOS7, do not forget to call
    // [[SDImageCache sharedImageCache] clearMemory];
    @autoreleasepool{
        // do not decode animated images
        if (image.images) { return image; }
    
        if (imageRef == nil) {
            return nil;
        }

        CGImageRef imageRef = image.CGImage;
    
        CGImageAlphaInfo alpha = CGImageGetAlphaInfo(imageRef);
        BOOL anyAlpha = (alpha == kCGImageAlphaFirst ||
                         alpha == kCGImageAlphaLast ||
                         alpha == kCGImageAlphaPremultipliedFirst ||
                         alpha == kCGImageAlphaPremultipliedLast);
    
        if (anyAlpha) { return image; }
    
        size_t width = CGImageGetWidth(imageRef);
        size_t height = CGImageGetHeight(imageRef);
    
        // current
        CGColorSpaceModel imageColorSpaceModel = CGColorSpaceGetModel(CGImageGetColorSpace(imageRef));
        CGColorSpaceRef colorspaceRef = CGImageGetColorSpace(imageRef);
        
        bool unsupportedColorSpace = (imageColorSpaceModel == 0 || imageColorSpaceModel == -1 || imageColorSpaceModel == kCGColorSpaceModelCMYK || imageColorSpaceModel == kCGColorSpaceModelIndexed);
        if (unsupportedColorSpace)
            colorspaceRef = CGColorSpaceCreateDeviceRGB();
    
        CGContextRef context = CGBitmapContextCreate(NULL, width,
                                                     height,
                                                     CGImageGetBitsPerComponent(imageRef),
                                                     0,
                                                     colorspaceRef,
                                                     kCGBitmapByteOrderDefault | kCGImageAlphaPremultipliedFirst);
    
        // Draw the image into the context and retrieve the new image, which will now have an alpha layer
        CGContextDrawImage(context, CGRectMake(0, 0, width, height), imageRef);
        CGImageRef imageRefWithAlpha = CGBitmapContextCreateImage(context);
        UIImage *imageWithAlpha = [UIImage imageWithCGImage:imageRefWithAlpha scale:image.scale orientation:image.imageOrientation];
    
        if (unsupportedColorSpace)
            CGColorSpaceRelease(colorspaceRef);
        
        CGContextRelease(context);
        CGImageRelease(imageRefWithAlpha);
        
        return imageWithAlpha;
    }
}
```

**Comment by mythodeia** [comment]
@sun-fox-cj please enable the CG_CONTEXT_SHOW_BACKTRACE argument in the environment variable in your scheme so we can understand why the image is showing as nil.
My best guess is that it has to do with the CGImageGetBitsPerComponent(imageRef).
check [this](https://github.com/rs/SDWebImage/issues/1401)

**Comment by sun-fox-cj** [comment]
@mythodeia I think the url should be checked, because the reason why I got this error is my url has been redirected, so I propose checked the HTTP status code first and then checked the image's size. if the HTTP status code is not 200 or image's size is zero, it's a bad image

**Comment by yrx0910** [comment]
I have met the same problem.    I  add the code at the first line of this method.
if (!image) {
            return nil;
        }

**Comment by sun-fox-cj** [comment]
@yrx0910 yes, it can work, but my team uses pod, I can't fix it in the file, so I have to write category, I hope it can be fixed in the source code

**Comment by bpoplauschi** [comment]
It's already fixed, in version 3.7.6. Just do a pod update
